src/player.cpp

```diff
@@ -11083,25 +11143,25 @@ void player::forced_dismount()
     } else {
         add_msg( m_debug, "Forced_dismount could not find a square to deposit player" );
     }
-    if( g->u.get_grab_type() != OBJECT_NONE ) {
-        add_msg( m_warning, _( "You let go of the grabbed object." ) );
-        g->u.grab( OBJECT_NONE );
+    if( is_player() ) {
+        if( g->u.get_grab_type() != OBJECT_NONE ) {
+            add_msg( m_warning, _( "You let go of the grabbed object." ) );
+            g->u.grab( OBJECT_NONE );
+        }
+        set_movement_mode( PMM_WALK );
+        g->update_map( g->u );
     }
     moves -= 150;
-    set_movement_mode( PMM_WALK );
-    g->update_map( g->u );
 }
 
 void player::dismount()
 {
     if( is_mounted() ) {
         if( const cata::optional<tripoint> pnt = choose_adjacent( _( "Dismount where?" ) ) ) {
-            if( g->is_empty( *pnt ) ) {
-                tripoint temp_pt = *pnt;
-                int xdiff = pos().x - temp_pt.x;
-                int ydiff = pos().y - temp_pt.y;
+            if( g->is_empty( *pnt ) && mounted_creature ) {
```

Comment:
```suggestion
            if( g->is_empty( *pnt ) ) {
```
The check for `mounted_creature` is already included in the check for `is_mounted()` above. No need to repeat it here.

Reply:
Btw. you can make functions like this much better to read by using early returns:
```C++
if( !some_condition ) {
    add_msg(" ... "); // if needed
    return;
}
const foo = ...; // calculate something
if( !some_other_condition ) {
    return;
}

// finally do the actual stuff here
```

The main code of you function (the actual stuff) is not intended that much. In this case, you can save three levels of indentation.